symbol.md

```diff
@@ -0,0 +1,181 @@
+  soldiers.collect(&:salary) #=> [4000, 5000, 8000]
+
+  #to get soldiers with grade_a_weapon
+  soldiers.select(&:grade_a_weapon?) #=> [#<Context::Soldier:0x000000017065b0 @name="Big Mike", @weapon="Bazooka">, #<Context::Soldier:0x00000001706538 @name="Small John", @weapon="Pistol">]
+  soldiers.select(&:grade_a_weapon?).map(&:weapon) #=> ["Bazooka", "Pistol"]
+```
+_Here's one for you to try out: write code to retrieve the last_name of soldiers with grade_a_weapon. Share your implementation in the comment section._
+
+Now let's match towards a more ambitious adventure.
+
+We can override the `to_proc` method in the Symbol class to achieve custom result. For instance say we want to always append the team captain's name to all `to_proc` requests;
+
+```ruby
+  class Symbol
+    def to_proc
+      Proc.new { |x| "#{x.send(self)} (signed John Doe Commander)" }
+    end
+  end
+
+  # Then we have
+  soldiers.map(&:name) #=> ["Big Mike (signed John Doe Commander)", "Small John (signed John Doe Commander)", "Lightening Larry (signed John Doe Commander)"]
+```
+Of course this will override every other symbol `to_proc` conversion
+
+```ruby
+  [1, 2, 3, 4, 5].map(&:to_f) #=> ["1.0 (signed John Doe Commander)", "2.0 (signed John Doe Commander)", "3.0 (signed John Doe Commander)", "4.0 (signed John Doe Commander)", "5.0 (signed John Doe Commander)"]
+```
+
+What if we don't want to depend on the **Symbol** `to_proc` conversion, well, we can define our own.
+
+Say we want to still append the team captain's name to our conversion but we don't want it to override other `symbol#proc` conversion, we can create a custom class with a to_proc method defined therein and then pass that as the arg to our map method method.
+
+```ruby
+  class CustomToProc
+    def self.to_proc
+      Proc.new { |x| "#{x.name} (signed John Doe Commander)" }
+    end
+  end
+
+  #Then we pass our custom to the map method
+  soldiers.map(&CustomToProc) #=> ["Big Mike (signed John Doe Commander)", "Small John (signed John Doe Commander)", "Lightening Larry (signed John Doe Commander)"]
+
+  # And our default symbol#to_proc remains unchanged
+  [1, 2, 3, 4, 5].map(&:to) #=> [1.0, 2.0, 3.0, 4.0, 5.0]
+```
+
+One common gotchas to watch out for is that symbol#to_proc does not take an extra argument
+```ruby
+  # say you want to retrieve the names of all the grade_a_weapon?
+  soldiers.select(&:grade_a_weapon?.name) #=> undefined method `name' for :grade_a_weapon?:Symbol
+
+  # or
+  soldiers.select(&:grade_a_weapon?, name) #=> syntax error, unexpected ',', expecting ')'
+
+```
+
+## Conclusion
+Symbol to proc are really useful and often come in handy in the drive for cleaner concise codes. One rule of thumb for when to use symbol to proc is whenever you find yourself having blocks whose sole function is call a single method on the passed arg i.e
+```ruby
+  something {|i| i.foo }
+
+  # should be replaced with
+  something(&:foo)
+```
+examples
+```ruby
+  (1..100).reject{|num| num.odd? }
```

Comment:
Can you show your readers the alternative implementation for each of this example using `Symbols#to_proc`?